src/vivarium/framework/randomness/index_map.py

```diff
@@ -192,7 +192,11 @@ def _hash(self, keys: pd.Index[Any], salt: ClockTime = 0) -> pd.Series[int]:
         return new_map % len(self)
 
     def _convert_to_ten_digit_int(
-        self, column: pd.Series[datetime | int | float]
+        self,
+        column: pd.Series[pd.Timestamp]
```

Comment:
This was typed incorrectly before but basically to make myppy happy, we have to have pd.Timestamp as the type to include datetime because pandas always converts datetimes to Timestamps. However, Timestamp is technically a part of datetime. See: https://pandas.pydata.org/docs/user_guide/timeseries.html

Reply:
So even though pd.Timestamp inherits from datatime, you have to include it explicitly (despite _also_ having datetime as part of the signature)?

Also, glad you caught that incorrect mashing of series types!

Reply:
Yes - I think technically we could just make it pd.Timestamp and not have a series as datetime as an options since no matter what you do pandas will convert it but I t hink this is more intuitive since we do use datetime elsewhere and we have Time in types.py which is a Timestamp | datetime (and unfortunately I could not do a serious of that).